typescript/schedules.md

```diff
@@ -0,0 +1,1157 @@
+    return await this.connection.withDeadline(deadline, fn);
+  }
+
+  /**
+   * Set metadata for any service requests executed in `fn`'s scope.
+   *
+   * @returns returned value of `fn`
+   *
+   * @see {@link Connection.withMetadata}
+   */
+  async withMetadata<R>(metadata: Metadata, fn: () => Promise<R>): Promise<R> {
+    return await this.connection.withMetadata(metadata, fn);
+  }
+
+  /**
+   * Create a new Schedule.
+   *
+   * @throws {@link ScheduleAlreadyRunning} if there's a running (not deleted) Schedule with the given `id`
+   */
+  public async create<Action extends ScheduleActionType>(options: ScheduleOptions<Action>): Promise<ScheduleHandle> {  }
+
+  /**
+   * List Schedules with an `AsyncIterator`:
+   *
+   * ```ts
+   * for await (const schedule: Schedule of client.list()) {
+   *   const { id, memo, searchAttributes } = schedule
+   *   // ...
+   * }
+   * ```
+   *
+   * To list one page at a time, instead use the raw gRPC method {@link WorkflowService.listSchedules}:
+   *
+   * ```ts
+   * await { schedules, nextPageToken } = client.scheduleService.listSchedules()
+   * ```
+   */
+  public list(options?: ListScheduleOptions): AsyncIterator<ListScheduleEntry> {}
+
+  /**
+   * Get a handle to a Schedule
+   *
+   * This method does not validate `scheduleId`. If there is no Schedule with the given `scheduleId`, handle
+   * methods like `handle.describe()` will throw a {@link ScheduleNotFoundError} error.
+   */
+  public getHandle(scheduleId: string): ScheduleHandle {  }
+}
+````
+
+### Higher-level TS client
+
+```ts
+import { Client } from '@temporalio/client'
+
+const client = new Client(options)
+
+client.schedule.create()
+client.workflow.start()
+client.asyncCompletion.heartbeat()
+client.operator.addSearchAttributes()
+
+interface ClientOptions {
+  dataConverter?: DataConverter
+  interceptors?: {
+    workflow: WorkflowClientInterceptors
+    schedule: ScheduleClientInterceptors
+  }
+  identity?: string
+  connection?: ConnectionLike
+  namespace?: string
+  queryRejectCondition?: temporal.api.enums.v1.QueryRejectCondition
+}
+```
+
+### Interceptors
+
+```ts
+interface ScheduleClientCallsInterceptor {
+  /**
+   * Intercept a service call to CreateSchedule
+   */
+  create?: (
+    input: ScheduleStartInput,
+    next: Next<this, 'create'>
+  ) => Promise<string /* conflictToken */>
+}
+
+interface ScheduleClientCallsInterceptorFactoryInput {
+  id: string
+}
+
+/**
+ * A function that takes a {@link ScheduleClientCallsInterceptorFactoryInput} and returns an interceptor
+ */
+export interface ScheduleClientCallsInterceptorFactory {
+  (
+    input: ScheduleClientCallsInterceptorFactoryInput
+  ): ScheduleClientCallsInterceptor
+}
+
+/**
+ * A mapping of interceptor type of a list of factory functions
+ */
+export interface ScheduleClientInterceptors {
+  calls?: ScheduleClientCallsInterceptorFactory[]
+}
+```
+
+- cron_string holds a traditional cron specification as a string. It
```

Comment:
remove this (you copied it above already)

Reply:
thanks!